test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -952,32 +883,28 @@ def test_max_jobs_analyze_limit_ramp_up(self):
         self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '4'})
         a = {"power_ramp_rate_enable": True,
              'max_jobs_analyze_limit': '2', 'node_idle_limit': '30'}
-        self.modify_hook_config(attrs=a, id='PBS_power')
+        self.modify_hook_config(attrs=a, hook_id='PBS_power')
         a = {'freq': 60}
         self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
                             sudo=True)
         a = {'enabled': 'True'}
         self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
                             sudo=True)
-        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '10'})
-        self.logger.info("Waiting for 12 mins to ramp down all the nodes")
-        time.sleep(720)
+        self.logger.info("Waiting for 15 mins to ramp down all the nodes")
+        time.sleep(900)
         for n in self.server.status(NODE):
             if 'resources_available.PBScraynid' in n:
-                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
+                if n['state'] != 'offline':
+                    self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
```

Comment:
Consider turning this into an assert().  This is unless you think the node might take some time to go into the sleep state.  Using expect() will continue checking for 30s.